base/generator.jl

```diff
@@ -108,11 +108,50 @@ Base.HasEltype()
 iteratoreltype(x) = iteratoreltype(typeof(x))
 iteratoreltype(::Type) = HasEltype()  # HasEltype is the default
 
+abstract type IteratorAccess end
+struct ForwardAccess <: IteratorAccess end
```

Comment:
Will folks want `ReverseAccess`? In other words, is this intended to collect all supported/efficient ways of iterating over an object? If so we may need to think about traits for iterator arithmetic, etc.
- https://en.cppreference.com/w/cpp/iterator/iterator_tags
- https://en.cppreference.com/w/cpp/iterator/iterator_traits

Reply:
The idea was indeed to also support `ReverseAccess` (although I can't read my mind from almost 3y ago!) and this was definitely inspired by the c++ traits. Cf. my [comment](https://github.com/JuliaLang/julia/pull/22489#issuecomment-596223857), `ReverseAccess` is probably "orthogonal" to the three access patterns currently included here.